pkg/apis/core/v1/conversion.go

```diff
@@ -270,6 +270,17 @@ func Convert_v1_PodStatus_To_core_PodStatus(in *v1.PodStatus, out *core.PodStatu
 			},
 		}
 	}
+
+	// If both fields (v1.HostIPs and v1.HostIP) are provided and differ, then HostIP is authoritative for compatibility with older kubelets
+	if (len(in.HostIP) > 0 && len(in.HostIPs) > 0) && (in.HostIP != in.HostIPs[0].IP) {
+		out.HostIPs = []core.HostIP{{IP: in.HostIP}}
+	}
+	// at the this point, autoConvert copied v1.HostIPs -> core.HostIPs
+	// if v1.HostIPs was empty but v1.HostIP is not, then set core.HostIPs[0] with v1.HostIP
+	if len(in.HostIP) > 0 && len(in.HostIPs) == 0 {
+		out.HostIPs = []core.HostIP{{IP: in.HostIP}}
```

Comment:
I think this is populating an alpha field, even when the feature is not enabled and will drop it on write... will this cause churn in kubelet seeing diffs in pod status and thinking it needs to reconcile? https://github.com/kubernetes/kubernetes/pull/88988 comes to mind

Reply:
yeah, confirmed in https://github.com/kubernetes/kubernetes/issues/109146#issuecomment-1083119908

Reply:
Add a test for this?